app/src/main/cpp/skyline/loader/nso.cpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include <lz4.h>
+#include <nce.h>
+#include <os.h>
+#include <kernel/memory.h>
+#include "executable.h"
+#include "nso.h"
+
+namespace skyline::loader {
+    NsoLoader::NsoLoader(const std::shared_ptr<vfs::Backing> &backing) : Loader(backing) {
+        u32 magic{};
+        backing->Read(&magic);
+
+        if (magic != util::MakeMagic<u32>("NSO0"))
+            throw exception("Invalid NSO magic! 0x{0:X}", magic);
+    }
+
+    std::vector<u8> NsoLoader::GetSegment(const std::shared_ptr<vfs::Backing> &backing, NsoSegmentHeader &segment, u32 compressedSize) {
```

Comment:
Same as `NroLoader::GetSegment`, the `segment` argument can be made into a `const` reference